storage/database/db_manager.go

```diff
@@ -1885,20 +1843,44 @@ func (dbm *databaseManager) ReadPreimageFromOld(hash common.Hash) []byte {
 	return data
 }
 
+func (dbm *databaseManager) WriteTrieNode(hash common.Hash, node []byte) error {
+	dbm.lockInMigration.RLock()
+	defer dbm.lockInMigration.RUnlock()
+
+	if dbm.inMigration {
+		if err := dbm.getDatabase(StateTrieMigrationDB).Put(TrieNodeKey(hash), node); err != nil {
+			return err
+		}
+	}
+	return dbm.getDatabase(StateTrieDB).Put(TrieNodeKey(hash), node)
+}
+
+func (dbm *databaseManager) PutTrieNodeToBatch(batch Batch, hash common.Hash, node []byte) error {
+	return batch.Put(TrieNodeKey(hash), node)
+}
+
 // WritePreimages writes the provided set of preimages to the database. `number` is the
 // current block number, and is used for debug messages only.
-func (dbm *databaseManager) WritePreimages(number uint64, preimages map[common.Hash][]byte) {
+func (dbm *databaseManager) WritePreimages(number uint64, preimages map[common.Hash][]byte) error {
 	batch := dbm.NewBatch(StateTrieDB)
 	for hash, preimage := range preimages {
 		if err := batch.Put(preimageKey(hash), preimage); err != nil {
-			logger.Crit("Failed to store trie preimage", "err", err)
+			return err
```

Comment:
Most of databaseManager methods are implemented to raise panic when it fails to write/delete something, and it returns an empty value when it fails to read something. I think the initial developer think the writing failure should be detected by a node operator, and don't want to miss any error handling that causes a critical database integrity issue. 
Is there any reason you made this exceptional case?

Reply:
@aidan-kwon, please refer to https://github.com/klaytn/klaytn/pull/1846#discussion_r1206227026.
the reason was to preserve the original behavior of statedb.Database.writeBatchPreimages().

However, it's not a good sign two people asked about the exact same line.

I quickly skimmed [db_manager.go](https://github.com/klaytn/klaytn/blob/v1.10.2/storage/database/db_manager.go) and discovered that some Klaytn-specific methods return errors. It's worth discussing whether we're going to use errorless interfaces for all DBManager methods.